### ISSUE TITLE
Map server Crash NPC_SELFDESTRUCTION

### ISSUE THREAD

**Issue by LadyNanuia** [rathena/rathena#8852]
### rAthena Hash

97a124c

### Client Date

2023-11-10

### Server Mode

Renewal

### Result

Map server crashes with this interaction it seems, i came back to a debug mode output message, ill post the screenshots below.
![image](https://github.com/user-attachments/assets/9f79388c-2740-404b-b37f-5b48758cbcdb)
![image](https://github.com/user-attachments/assets/99f3b804-3868-416e-83dd-a05f8bd51833)
![image](https://github.com/user-attachments/assets/61720267-53c0-42e3-902b-5ad10723e64b)

### Relevant Log Output

```Shell
posted above
```

### Expected Result

Map Server is not supposed to crash

### How to Reproduce

i was away when this happend but it only started happening after this latest merge as mentioned

### Official Information

It should work? i guess

### Modifications that may affect results

None

**Comment by dzurrahman** [comment]
yes i got a same problem, when u meat some npc with function `OnTouch`, u will freeze, after 5 minute, map will crash

**Comment by dzurrahman** [comment]
i fix this issue revirt some commit from [a58584c3ff5b67f349ccad1d3263853917285d15](https://github.com/rathena/rathena/commit/a58584c3ff5b67f349ccad1d3263853917285d15)

**Comment by Daegaladh** [comment]
I can't reproduce your issue, is there any steps to cause the crash? I've tested with both single and chained self destructions

**Comment by dzurrahman** [comment]
@Daegaladh i have same issue some npc make crash, then i revert commit, u can check try to warp into prt_fild01 137 367 and go inside, the requirment is u need have complete quest illusion daily episode 17.1 illusion, i dont know if my issue same as him. thanks

**Comment by dzurrahman** [comment]
> > @Daegaladh i have same issue some npc make crash, then i revert commit, u can check try to warp into prt_fild01 137 367 and go inside, the requirment is u need have complete quest illusion daily episode 17.1 illusion, i dont know if my issue same as him. thanks
> 
> Sorry but what does that have to do with NPC_SELFDESTRUCTION?

i dont know exactly, but the reference error for me is the same as this picture
![image](https://github.com/user-attachments/assets/82f1280d-081b-4c69-94dc-1083be9c9b0a)

**Comment by Lemongrass3110** [comment]
@LadyNanuia are you able to reproduce the crash?

Sadly you cut of the most important information on your screenshot:
![grafik](https://github.com/user-attachments/assets/31cf2823-ca8d-463e-a831-0750e86df108)

skill.cpp:13729 is a commented line at master: `//if(skill_lv <= 0) return 0;`

**Comment by LadyNanuia** [comment]
> @LadyNanuia are you able to reproduce the crash?
> 
> Sadly you cut of the most important information on your screenshot: ![grafik](https://private-user-images.githubusercontent.com/3517879/391561281-31cf2823-ca8d-463e-a831-0750e86df108.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNjIxNDgsIm5iZiI6MTczMzE2MTg0OCwicGF0aCI6Ii8zNTE3ODc5LzM5MTU2MTI4MS0zMWNmMjgyMy1jYThkLTQ2M2UtYTgzMS0wNzUwZTg2ZGYxMDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDJUMTc1MDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2Y4MTM4MzAzMTQwMDJjYjE2MWYyMTMwYjQ3ZTg5OGM5NmUyY2ZhODNjODc2MjYyNmI3OTQ1ZmVkZGIxZjlmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.21nZ9aepKFlhIktFjGzicBFKrqOg6EOIgY3VRylp1_w)
> 
> skill.cpp:13729 is a commented line at master: `//if(skill_lv <= 0) return 0;`

sadly not, it only happend once

**Comment by Beginerxxxxx** [comment]
> 
> Just merge rathena master into your fork, that's what Git is for... 🤦‍♂️

I can't use the all master file as it is. I need to update and insert rows into each file because my server can use all the skills of every class, and additional functions have been patched in."

So, I'm not sure which files might be inconsistent or which ones are using the functions that have changed.

My opinion is that the issue the poster encountered might be due to something being missed, or perhaps an update was made incorrectly.

**Comment by Daegaladh** [comment]
The only thing I've noticed is that we don't check for `mapdata->block_mob == nullptr` in map_foreachinrangeV, but that doesn't explain the real cause

**Comment by skstrife** [comment]
Hi, could you try the Lasagna mission, especially in steps 1 to 6 where classchange is used. I have done 5 tests and the map fails either in step 1 or step 6. @Lemongrass3110
